sdks/java/harness/src/main/java/org/apache/beam/fn/harness/data/PCollectionConsumerRegistry.java

```diff
@@ -213,38 +211,39 @@ public MetricTrackingFnDataReceiver(
       this.pTransformId = consumerAndMetadata.getPTransformId();
       HashMap<String, String> labels = new HashMap<String, String>();
       labels.put(Labels.PCOLLECTION, pCollectionId);
+
+      // Collect the metric in a metric container which is not bound to the step name.
+      // This is required to count elements from impulse steps, which will produce elements outside
```

Comment:
I appreciate the detailed comment, but I still don't actually understand what it means :-(

Reply:
I don't really either, I just moved it from below.

Maybe we can do better and we can somehow ensure it *only* happens for impulse steps